in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,265 @@
+                    "<sign | verify>\n{0}"
+                    "[--replaceall]\n{0}"
+                    "[--infixkeyid]\n{0}"
+                    "<path/to/signable>"
+                    "[--verbose]\n\n"
+                    .format(lpad))
+
+    in_toto_args = parser.add_argument_group("in-toto-sign options")
+
+    in_toto_args.add_argument("-k", "--key", type=str, required=True,
+                              help="Path to private key to sign link metadata (PEM)")
+
+    in_toto_args.add_argument("operator", type=str, choices=['sign', 'verify'],
+                              help="sign or verify")
+
+    in_toto_args.add_argument("-r", "--replaceall", action="store_true",
+                              help = "Whether to replace all the old signatures"
+                              "or not")
+
+    in_toto_args.add_argument("-i", "--infixkeyid", action="store_true",
+                              help = "whether to infix keyID in the file name"
+                              "or not")
+
+    in_toto_args.add_argument("signablepath", type=str,
+                              help="path to the signable file")
+
+    in_toto_args.add_argument("-v", "--verbose", dest="verbose",
+                              help="Verbose execution.", default=False,
+                              action="store_true")
+
+    args = parser.parse_args()
+    args.operator = args.operator.lower()
+
+    return args
+
+
+def main():
+    """
+    First calls parse_args to parse the arguments, and then calls either
+    add_sign or add_replace depending upon the arguments. Based on the
+    arguments it then dumps the corresponding file.
+
+    """
+    args = parse_args()
+    rsa_key = in_toto.util.import_rsa_key_from_file(args.key)
+
+    if args.verbose:
+        log.logging.getLogger.setLevel(log.logging.INFO)
+
+    if args.operator == 'sign':
+
+      try:
+        if not args.replaceall:
+
+          if not args.infixkeyid:
+            signable_object = add_sign(args.signablepath, args.key)
+            signable_object.dump()
+            sys.exit(0)
+
+          else:
+            signable_object = add_sign(args.signablepath, args.key)
+            signable_object.dump(key = rsa_key)
+            sys.exit(0)
+
+        else:
+
+          if not args.infixkeyid:
+            signable_object = add_replace(args.signablepath, args.key)
+            signable_object.dump()
+            sys.exit(0)
+
+          else:
+            signable_object = add_replace(args.signablepath, args.key)
+            signable_object.dump(key = rsa_key)
+            sys.exit(0)
+
+      except Exception as e:
+        print('The following error occured while signing', e)
+        sys.exit(2)
+
+    elif args.operator == 'verify':
+
+      try:
+        if verify_sign(args.signablepath, args.key):
+          sys.exit(0)
+        else:
+          sys.exit(1)
+
+      except Exception as e:
+        print('The following error occured while verification', e)
+        sys.exit(3)
```

Comment:
I"m not sure we are documenting these error codes are we? Is it worth having it something different on each case?